src/javascript/utils.js

```diff
@@ -253,6 +253,51 @@ function assignIfUndefined (subject, target) {
 	}
 }
 
+/**
+ * Used to find out all the paths which contain a circular reference.
+ * @param {*} rootObject The object we want to search within for circular references
+ * @returns {string[]} Returns an array of strings, the strings are the full paths to the circular references within the rootObject
+ */
+function findCircularPathsIn(rootObject) {
+	const traversedValues = new WeakSet();
```

Comment:
Users would need a new polyfill for `WeakSet`so this is a breaking change

Reply:
I changed the base branch to merge into v3 instead

Reply:
cool :) don't forget to add it to `origami.json`!